File: RecipeRandomizer.Business/Models/Identity/ChangePasswordRequest.cs
using System.ComponentModel.DataAnnotations;

namespace RecipeRandomizer.Business.Models.Identity
{
    public class ChangePasswordRequest
    {
        [Required]
        public int Id { get; set; }

        [Required]
        public string Password { get; set; }

        [Required]
        [MinLength(6)]
        public string NewPassword { get; set; }

        [Required]
        [Compare("NewPassword")]
        public string ConfirmNewPassword { get; set; }
    }
}

File: RecipeRandomizer.Web/ClientApp/src/app/models/identity/changePasswordRequest.ts
export class ChangePasswordRequest {
    password: string;
    newPassword: string;
    confirmNewPassword: string;

    constructor(password: string, newPassword: string, confirmNewPassword: string) {
        this.password = password;
        this.newPassword = newPassword;
        this.confirmNewPassword = confirmNewPassword;
    }
}
